Update account-functions.adoc with warning of invoke v0 txs to execute (#1298)
starknet-io/starknet-docs: components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc

Separating the validation and execution stages guarantees payment to sequencers for work completed and protects them from Denial of Service (DoS) attacks.

[#attacks_that_validation_limitations_prevent]

> == Potential attacks on `+__validate__+`

Reviewer: ```suggestion
== Potential attacks on validation functions
```

** `get_block_hash`
** `get_sequencer_address` `get_sequencer_address` is only supported for Cairo 0 contracts.

[#attacks_that_execute_limitations_prevent]

> == Potential attacks on `+__execute__+`

Reviewer: ```suggestion
== Potential attacks on the `+__execute__+` function
```

* Deprecated `INVOKE` v0 transactions can call the `+__execute__+` function of all contracts without going through `+__validate__+ first. 
* The `+__execute__+` function can be called by any contract.

[#ways_to_mitigate_execute_attacks]

> == Ways to mitigate attacks on `+__execute__+`

Reviewer: ```suggestion
== Ways to mitigate attacks on the `+__execute__+` function
```

* Account contracts should explicitly assert in the `+__execute__+` function that the transaction version is 1 or above.

> * Account contracts should explicitly assert in the `+__execute__+` function that the sender is 0 (i.e., the sequencer).

Reviewer: ```suggestion
* Account contracts should explicitly assert in the `+__execute__+` function that the sender's address is 0, that is, the sender is the sequencer.
```

Initiates the execution stage in the sequencer. The sequencer calls this function upon receiving an `INVOKE` transaction, after the `+__validate__+` function successfully completes.

[WARNING]
====
Deprecated `INVOKE` v0 transactions can call the `+__execute__+` function of all contracts. To mitigate this risk, `+__execute__+` calls should explicitly assert that the transaction version is 1 or above.
====

> 

Author: @stoobie not sure if the warning is in the correct format. Would also be good to link INVOKE v0 to this page: https://docs.starknet.io/architecture-and-concepts/network-architecture/transactions/#v0_deprecated_hash_calculation
